packages/@vue/cli-service/lib/commands/serve.js

```diff
@@ -89,7 +90,7 @@ module.exports = (api, options) => {
       protocol,
       host,
       port,
-      options.baseUrl
+      isAbsoluteUrl(options.baseUrl) ? '/' : options.baseUrl
```

Comment:
Won't that break for `baseUrl: http://localhost:8081/someBasePath/` ?

Reply:
Set baseUrl to `http://localhost:8081/someBasePath/` doesn't mean to start the dev-server on `localhost:8081`, the dev-server would still be started on `${host}:${port}`.

Reply:
Sure, but doesn't the console now say " go to host: port/" even though people would have to go to "localhost:port/baseurl"?

I'm honestly confused because paths and hosts in webpack can be confusing as hell to me

Reply:
Consider those two cases here:

1. Asset files (js/css/img) are served by `webpack-dev-server` (eg. localhost:8080), and user uses a reverse proxy to access the **html** (eg. localhost:3000/app/). In this case, `baseUrl` should be set to `http://localhost:8080/`, but user need to open `http://localhost:3000/app/` in browser.

2. Asset files (js/css/img) are served by `webpack-dev-server` (eg. localhost:8080), and user uses a reverse proxy to access the **asset files** (eg. localhost:3000/assets/). In this case, `baseUrl` should be set to `http://localhost:3000/assets/`, and user need to go to `http://localhsot:8080` in browser.

What I want to say is, we cannot infer which url should user open in browser if user set `baseUrl` to a absolute url. What we can do is tell user "we have started your dev-server on localhost:8080"... 

And, yes it is confusing. 😂

Reply:
Awesome, that was very helpful. Thanks.